### PR TITLE
Various: Fix typos

### PR DESCRIPTION
**Pull request by goggle**
### Description

This PR corrects various typos found in the repository.
Typos from `deps/` have not been included here, because I guess this is _external_ software?

### Motivation and Context

It's technically not required, but it might improve the code quality of the repository.

### How Has This Been Tested?

Other than running `clang-format` it has not been tested. 

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

File: libobs/obs-properties.h [hunk 1]
```diff
@@ -428,7 +428,7 @@ EXPORT const char *obs_property_button_url(obs_property_t *p);
 
 #ifndef SWIG
 OBS_DEPRECATED
-EXPORT enum obs_text_type obs_proprety_text_type(obs_property_t *p);
+EXPORT enum obs_text_type obs_property_text_type(obs_property_t *p);
```
Comment (notr1ch) [inline libobs/obs-properties.h]: This is an API break, the typoed function is intentionally there for backward compatibility and marked deprecated.

**Review by gxalpha — CHANGES_REQUESTED** [review]
Please don't change any existing API's, or other external headers in the repo.
Testing a PR before submitting it is also generally appreciated. This PR doesn't even compile.

**Review by notr1ch — COMMENTED** [review]
There are a number of changes to external header files and other non-OBS code that should not be included, these will be lost any time the headers or SDKs are updated. Some examples are the simde code, Intel Media SDK (libmfx), NVENC headers and Decklink.

The commit messages should be properly formatted, given the widespread changes, prefixing every module is a bit overkill so perhaps "Various: Fix typos" or similar would work.

**Comment by goggle** [comment]
Maybe someone could mention the files that should not be included here.
Here is a list of the currently changed files:

```
UI/media-controls.cpp
UI/media-controls.hpp
UI/obs-app.cpp
UI/window-basic-auto-config.cpp
UI/window-basic-settings-stream.cpp
UI/window-basic-settings.cpp
UI/youtube-api-wrappers.cpp
UI/youtube-api-wrappers.hpp
cmake/bundle/macOS/entitlements.plist
cmake/external/ObsPluginHelpers.cmake
cmake/finders/FindLibrnnoise.cmake
docs/sphinx/reference-properties.rst
libobs/obs-encoder.c
libobs/obs-source.h
libobs/obs.h
libobs/util/simde/simde-align.h
libobs/util/simde/simde-common.h
libobs/util/simde/simde-diagnostic.h
libobs/util/simde/simde-features.h
libobs/util/threading-windows.h
plugins/aja/aja-card-manager.hpp
plugins/decklink/linux/decklink-sdk/DeckLinkAPIConfiguration.h
plugins/decklink/mac/decklink-sdk/DeckLinkAPIConfiguration.h
plugins/decklink/mac/decklink-sdk/DeckLinkAPIStreaming.h
plugins/decklink/win/decklink-sdk/DeckLinkAPIConfiguration.idl
plugins/decklink/win/decklink-sdk/DeckLinkAPIStreaming.idl
plugins/linux-pipewire/pipewire.c
plugins/linux-pulseaudio/pulse-input.c
plugins/linux-pulseaudio/pulse-wrapper.h
plugins/linux-v4l2/v4l2-udev.c
plugins/mac-capture/CGDisplayStream.h
plugins/mac-virtualcam/src/common/MachProtocol.h
plugins/obs-ffmpeg/external/nvEncodeAPI.h
plugins/obs-ffmpeg/jim-nvenc-helpers.c
plugins/obs-filters/cmake/legacy.cmake
plugins/obs-filters/cmake/rnnoise.cmake
plugins/obs-filters/rnnoise/src/kiss_fft.c
plugins/obs-outputs/cmake/legacy.cmake
plugins/obs-outputs/librtmp/rtmp.c
plugins/obs-outputs/librtmp/rtmp.h
plugins/obs-qsv11/libmfx/include/mfx_dispatcher.h
plugins/obs-qsv11/libmfx/include/mfx_dispatcher_log.h
plugins/obs-qsv11/libmfx/include/mfx_exposed_functions_list.h
plugins/obs-qsv11/libmfx/include/mfx_library_iterator.h
plugins/obs-qsv11/libmfx/include/mfxaudio_exposed_functions_list.h
plugins/obs-qsv11/libmfx/include/msdk/include/mfxstructures.h
plugins/obs-qsv11/libmfx/src/main.cpp
plugins/obs-qsv11/libmfx/src/mfx_dispatcher.cpp
plugins/obs-qsv11/libmfx/src/mfx_dispatcher_log.cpp
plugins/obs-qsv11/libmfx/src/mfx_driver_store_loader.cpp
plugins/rtmp-services/service-specific/twitch.c
plugins/text-freetype2/find-font-iconv.c
plugins/win-dshow/win-dshow-encoder.cpp
```

**Comment by PatTheMav** [comment]
> Maybe someone could mention the files that should not be included here. 
 
```
libobs/util/simde/*
plugins/decklink/linux/decklink-sdk/*
plugins/decklink/mac/decklink-sdk/*
plugins/decklink/win/decklink-sdk/*
plugins/obs-ffmpeg/external/nvEncodeAPI.h
plugins/obs-filters/rnnoise/src/*
plugins/obs-outputs/librtmp/*
plugins/obs-qsv11/libmfx/*
```

That'd just be the external sources we include on first glance.

**Pull request closed without merging**